src/ngrest/base/ActiveButton.php

```diff
@@ -154,7 +155,29 @@ public function getCondition()
     {
         return empty($this->_condition) ? '' : $this->_condition;
     }
-
+    
+    private $_permissionLevel = null;
+    
+    /**
+     * Setter method for the ActiveWindow button displaying permission level
+     * @param int $permissionLevel
+     * @since 4.0.0
+     */
+    public function setPermissionLevel($permissionLevel)
+    {
+        $this->_permissionLevel = $permissionLevel;
+    }
+    
+    /**
+     * Get the button displaying permission level or empty of not set
+     * @inheritdoc
+     * @since 4.0.0
+     */
+    public function getPermissionLevel()
+    {
+        return $this->_permissionLevel === null ? null : $this->_permissionLevel;

```

Comment:
The oppsite would be less complicate:

```
return $this->_permissionLevel ? $this->_permissionLevel : null;
```

Or is there a need to explicit check for `null`?

Reply:
Permission level is an integer including 0 value (Auth::CAN_VIEW) so actually I am checking explicitly if it is null to know that it has not been initialized..